### PR TITLE
Feature/challenges

### PR DESCRIPTION
**Pull request by danielcfuentes**
## Description

what this pull request is doing:
This pull request creates challenges for users. The challenges are personalized as it gets the average of all the previous runs to create the challenges. Their is a cron job so that new challenges are created daily. If you do not complete the challenge during the time limit, it will mark it as failed. Once you complete a run, it updates the challenges and you can see the progress update of the challenges.

what will be done in later PRs and not included here:
I will go back and work on showing elevation on the map.
With the challenges i would like to later on add a leaderboard, streaks, and rewards if there is enough time.

## Milestones
Milestone: Challenges and Automation

## Resources
- https://www.npmjs.com/package/node-cron

## Test Plan
Screenshot 2024-07-16 at 11 42 17 PM

Screenshot 2024-07-17 at 12 02 19 AM
Screenshot 2024-07-17 at 12 02 06 AM
Screenshot 2024-07-17 at 12 01 37 AM

### CODE REVIEW

File: backend/deleteAllChallenges.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const { PrismaClient } = require("@prisma/client");
+const prisma = new PrismaClient();
+
+async function deleteAllChallenges() {
+  try {
+    const deletedChallenges = await prisma.challenge.deleteMany({});
+    console.log(`Successfully deleted ${deletedChallenges.count} challenges.`);
```
Comment (janicetsai1) [inline backend/deleteAllChallenges.js:7]: make sure to delete these console statements before merging!

File: frontend-runitup/src/Components/Recommendation/RecommendationPage.jsx [hunk 2]
```diff
@@ -1,12 +1,125 @@
...
+
+  const renderChallenges = () => {
+    if (loading) return <Spin size="large" />;
+    if (activeChallenges.length === 0) {
+      return (
+        <Empty
+          description={
+            <span>
+              No current challenges. <br />
+              Next challenges in: {timeUntilNextChallenge}
```
Comment (janicetsai1) [inline frontend-runitup/src/Components/Recommendation/RecommendationPage.jsx:93]: Nit but i assume only one challenge is generated so can update to singular form: Next challenge in

**Review by janicetsai1 — APPROVED** [review]
overall looks good. just make sure to address some nit comments and the merge conflict

File: frontend-runitup/src/Components/Recommendation/RecommendationPage.jsx (hunk 2 above)
Comment (danielcfuentes) [reply frontend-runitup/src/Components/Recommendation/RecommendationPage.jsx:93]: 3 Challenges are made at a time. And they are of different categories.

**Pull request merged**